grakn/rpc/database.py

```diff
@@ -0,0 +1,91 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+from abc import ABC, abstractmethod
+from typing import Set, Optional
+
+import grakn_protocol.protobuf.database_pb2 as database_proto
+
+from grakn.rpc.cluster.server_address import ServerAddress
+from grakn.rpc.utils import rpc_call
+
+
+class Database(ABC):
+
+    @abstractmethod
+    def name(self) -> str:
+        pass
+
+    @abstractmethod
+    def delete(self) -> None:
+        pass
+
+    class Replica(ABC):
+
+        @abstractmethod
+        def database(self) -> "DatabaseCluster":
+            pass
+
+        @abstractmethod
+        def term(self) -> int:
```

Comment:
Previously this data was hidden from the user; now it is exposed, for troubleshooting purposes.